az-core/src/main/java/azkaban/metrics/MetricsManager.java

```diff
@@ -120,11 +120,8 @@ public synchronized void startReporting(final String reporterName, final Props p
     if (metricsReporterClassName != null && metricsServerURL != null) {
       try {
         log.info("metricsReporterClassName: " + metricsReporterClassName);
-        final Class metricsClass = Class.forName(metricsReporterClassName);
-
-        final Constructor[] constructors = metricsClass.getConstructors();
-        constructors[0].newInstance(reporterName, this.registry, metricsServerURL);
-
+        final Class<?> metricsClass = Class.forName(metricsReporterClassName);
+        Utils.callConstructor(metricsClass, reporterName, this.registry, metricsServerURL);
```

Comment:
Thanks for improving the initialization of metrics! 
Since we're at it :) , could you please add a 4th **boolean** parameter to the constructor? The signature of the constructor we should use is:
 `public AmfReporter(String groupName, MetricRegistry registry, String amfServerUrl, boolean reportCounterAsGauge) {..}`

The parameter reportCountersAsGauge should be **true**.

Reply:
Added. Unfortunately, the Utils helper that I tried initially can't handle primitive types, so it's back to fetching and calling Ctor explicitly. If we want to do anything smarter (Ctor that takes context object? builder instead of Ctor?), let's do it separately. @ypadron-in

Reply:
I agree, thanks.